### PR TITLE
Remove `#[derive(Debug)]` from library structs

### LINKED ISSUE #326: Remove the #[derive(Debug)] on library structs
Placing a `#[derive(Debug)]` on user facing library structs that have generics puts an unnecessary burden on users of the library because now the user's type must implement `Debug` even if it has no business doing so.

It's completely fine to `#[derive(Debug)]` on structs that are not generic and/or hold instances of types constructed by users.

Is there any particular reason why the structs in `rpc/server.rs` for example derive Debug?

### PR DESCRIPTION
**Pull request by chansuke**
Fixes #326.

### CODE REVIEW

File: tarpc/src/rpc.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 // license that can be found in the LICENSE file or at
 // https://opensource.org/licenses/MIT.
 
-#![deny(missing_docs, missing_debug_implementations)]
```
Comment (tikue) [inline tarpc/src/rpc.rs:7]: Let's leave this in

File: tarpc/src/rpc/server.rs [hunk 2]
```diff
@@ -236,6 +240,12 @@ where
     }
 }
 
+impl<Req, Resp, T> fmt::Debug for BaseChannel<Req, Resp, T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "NewClient {:?}", self)
```
Comment (tikue) [inline tarpc/src/rpc/server.rs:245]:
```suggestion
        write!(f, "BaseChannel")
```

**Review by tikue — CHANGES_REQUESTED** [review]

File: tarpc/src/rpc.rs (hunk 1 above)
Comment (chansuke) [reply tarpc/src/rpc.rs]: Ok!

**Comment by chansuke** [comment]
@tikue Thank you for the review.

**Comment by tikue** [comment]
Woo! Thank you!

**Pull request merged**